### PR TITLE
[Feature] Plugin Script

### LINKED ISSUE #16: Plugin for sites not using Jekyll
Tl;dr: I want to add a custom JavaScript "plugin" that can easily add Primer Spec to HTML websites without having to use Jekyll. Where should such a script live?

---

The _long_ version of the context:

Dr. Kapritsos expressed interest in using Primer Spec with EECS 482 specs. However, their setup involves hosting plain HTML files on a private web server accessible only to students enrolled in the course.
I wrote a quick "plugin" script that adds the scaffolding that Jekyll would have otherwise provided. To demonstrate, I uploaded the Primer Spec theme files in a [demo GitHub repo](https://github.com/seshrs/ci-playground/tree/master/docs/assets) (and also uploaded the plugin code), and then created two test pages [`plain.html`](https://seshrs.ml/beta/plain.html) and [`spec.html`](https://seshrs.ml/beta/spec.html). The two pages are identical except that the plugin is added as a `script` tag at the end of `spec.html`.

Where should this plugin live? Here are two options I thought of:

### Option 1
Add custom plugin as an integration option built into Primer Spec.

*Pros:*
  - Plugin code would live in this repo and would share code with the Jekyll theme (so they would always be in sync).
  - Our current Jekyll theme can be modified to completely use the custom plugin. This could slightly reduce complexity introduced by Jekyll. This would also cause the Jekyll layout `spec.html` to look more similar to the upstream (GitHub's original Primer theme), possibly making maintenance easier.

*Cons:*
  - Added complexity. There will be more moving pieces.
  - We would have to generate two JavaScript targets.

### Option 2
Custom plugin is maintained in a separate repository.

_Pros:_
  - Separation of concerns. This repo would be dedicated as a Jekyll Theme.
  - The plugin (which would not be used by EECS 485) would not necessarily have to be maintained by `eecs485staff`.

_Cons:_
  - Additional maintenance work might be needed to keep the external plugin up-to-date with the Jekyll theme.
  - Concerns about who controls and maintains the custom plugin...?

---

I initially started writing the plugin with option (1) in mind, but I'm beginning to lean towards option (2). @awdeorio what do you think? And who should own the plugin code?

### PR DESCRIPTION
**Pull request by seshrs**
This PR makes Primer Spec available to non-Jekyll websites. Such websites would simply have to add a single script, which adds all the scaffolding, style and scripts needed to make Primer Spec render on that web page. (See #16 for context about why this feature could be useful.)

To minimize code duplication (and simplify the Jekyll theme code), `spec.html` layout is modified to simply use the plugin to generate the necessary scaffolding.

The PR also adds a Git pre-commit hook to ensure that each commit adds the most up-to-date production build of the JavaScript targets `spec_main.js` and `primer_spec_plugin.js`.

Finally, this PR cleans up the README, moves documentation to other docs files (like CONTRIBUTING.md) and adds instructions on how to integrate the theme or plugin.

/CC @ekaprits

Closes #16.

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+PROJECT_ROOT=$(dirname `dirname $0`)
```
Comment (awdeorio) [inline .githooks/pre-commit]: The best way to get the directory containing a script is this:
```bash
cd $(dirname $0) && pwd -P
```

File: .githooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+set -Euexo pipefail
```
Comment (awdeorio) [inline .githooks/pre-commit]: Should put this before the `PROJET_ROOT` stuff

File: .githooks/pre-commit [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+npx webpack --env prod
+
+git add $PROJECT_ROOT/assets/js/primer_spec_plugin.js
```
Comment (awdeorio) [inline .githooks/pre-commit]: It's not clear if these are minified/uglified.  I believe that they should be.  If so, consider using the common `X.min.js`.

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,9 +1,22 @@
 # Contributing to The Primer Spec Theme
 
-*This document was adapted from the corresponding documents from the original [Primer theme](https://github.com/pages-themes/primer).*
-
-Following these guidelines helps to communicate that you respect the time of the developers managing and developing this open source project. In return, they should reciprocate that respect in addressing your issue, assessing changes, and helping you finalize your pull requests.
-
+*This document was adapted in part from the corresponding documents from the original [Primer theme](https://github.com/pages-themes/primer).*
+
+## Contents
+- [Support](#looking-for-support)
```
Comment (awdeorio) [inline docs/CONTRIBUTING.md]: Nice

File: docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -22,48 +35,140 @@ Here are a few tips for writing *great* bug reports:
...
+
+4. Draft a new "release". Visit the [releases page](https://github.com/eecs485staff/primer-spec/releases). Note the most recent release's version number (for example, `1.0.0+fa19`).
+
+5. Decide what the next version number should be. If there are no major changes, simply bump the "patch version number" (for example, `1.4.1` would be followed by `1.4.2`). If there are design changes, bump the "minor version number" (for example, `1.4.1` would become `1.5.0`). Also specify the upcoming semester after the `+` symbol — this is metadata and is not parsed as part of the version number. (For more about versioning, see [Semver](https://semver.org/).)
+
+6. Click the "Draft a new release" button. Specify the version number. Title and description are optional. *(Switch the "target branch" to `master`. That said, the two branches should be in sync at the time of release so this should not really matter.)*
+
 
-At a high level, [the process for proposing changes](https://guides.github.com/introduction/flow/) is:
+## Modifications from Primer
```
Comment (awdeorio) [inline docs/CONTRIBUTING.md]: This is really helpful!

**Comment by seshrs** [comment]
> One question: does the work flow change for existing GitHub pages (Jekyll) users?

No, the workflow stays the same. Primer Spec can still be used as a remote theme exactly like before. When the final Jekyll site is built by GitHub for deployment, the JavaScript files are copied to the site's `assets` directory, just like before.

File: .githooks/pre-commit (hunk 3 above)
Comment (seshrs) [reply .githooks/pre-commit]: While it's true that they're uglified and minified in production, that is not the case in development. That said, I guess it's not a big deal to call them "minified"  — that's something that can be fixed later, if necessary.

**Pull request merged**